File: 0206-reverse-linked-list/0206-reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* nouvListe = nullptr;
        ListNode* iterator =head ;
        int nb=0;
        while(iterator != nullptr){
            ListNode * nouvElem = new ListNode(iterator->val,nouvListe);
            nouvListe=nouvElem ;
            iterator=iterator->next;
        }
        return nouvListe;
         
    }
};

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        it = head
        prev = None
        while it:
            temp = it.next
            it.next = prev
            prev = it
            it = temp
        return prev